ui/lib/config-ui/addon/components/messages/page/create-and-edit-message-form.js

```diff
@@ -38,15 +40,25 @@ export default class MessagesList extends Component {
     }
   }
 
+  get hasSomeActiveModals() {
+    if (!this.args.messages) return false;
+    return (
+      this.args.messages?.some((message) => message.type === 'modal' && message.active) &&
+      this.args.message.type === 'modal'
+    );
+  }
+
   @task
   *save(event) {
     event.preventDefault();
     try {
-      const { isValid, state, invalidFormMessage } = this.args.message.validate();
-      this.modelValidations = isValid ? null : state;
-      this.invalidFormAlert = invalidFormMessage;
+      let isValid;
 
-      if (isValid) {
+      if (!this.hasSomeActiveModals) {
+        isValid = this.validateMessageForm();
+      }
+
+      if (isValid || this.hasSomeActiveModals) {
```

Comment:
I'm not sure if this conditional has the functionality we want. If the form is invalid, but the user has some active modals the save will still continue, which we don't want, right? A save shouldn't not be attempted if there is any validation error.

Reply:
+1 on this point, I think the save should not continue if validations occur OR if there are active modals. I think the flow of the save method should be:

1. check validity of form
2. if invalid, abort save
3. check active modals
4. if active modals present, show modal & wait for user confirmation
5. if user selects continue, continue with save. Otherwise, exit save

The waiting for user confirmation piece would be good to have as a separate concurrency task that can just be yielded within the save task. Let me know if you want any guidance on how that should work!

Reply:
sounds good!